galsim/compound.py

```diff
@@ -902,3 +905,263 @@ def __setstate__(self, d):
+    @param  flux                    Optional total flux in all point sources.
+                                    [default: 1]
+    @param  rng                     Optional random number generator. Can be
+                                    any galsim.BaseDeviate.  If not sent or
+                                    None, the rng is created internally.
+                                    [default: None]
+    @param  gsparams                Optional GSParams for the gaussians
+                                    representing each point source.
+                                    [default: None]
+
+    Methods
+    -------
+
+    This class inherits from galsim.Sum. Additional methods are
+    
+        calculateHLR:
+            Calculate the actual half light radius of the generated points
+
+    There are also "getters",  implemented as read-only properties
+
+        .npoints
+        .input_half_light_radius
+        .flux
+        .gaussians
+            The list of galsim.Gaussian objects representing the points
+        .points
+            The array of x,y offsets used to create the point sources
+
+    Notes
+    -----
+
+    - The algorithm is a modified version of that presented in
+
+          https://arxiv.org/abs/1312.5514v3.
+
+      Modifications are
+        1) there is no outer cutoff to how far a point can wander
+        2) We use the approximation of an infinite number of steps.
+    """
+
+    # these allow use in a galsim configuration context
+
+    _req_params = { "npoints" : int, "half_light_radius" : float }
+    _opt_params = { "flux" : float }
+    _single_params = []
+    _takes_rng = True
+
+    def __init__(self, npoints, half_light_radius, flux=1.0, rng=None, gsparams=None):
+
+        self._half_light_radius = float(half_light_radius)
+
+        self._flux    = float(flux)
+        self._npoints = int(npoints)
+
+        # size of the galsim.Gaussian objects to use as delta functions
+        self._gaussian_sigma = 1.0e-8
+
+        self._input_gsparams=gsparams
+
+        # we will verify this in the _verify() method
+        if rng is None:
+            rng = galsim.BaseDeviate()
+
+        self._rng=rng
+
+        self._verify()
+
+        self._set_gaussian_rng()
+
+        self._points = self._get_points()
+        self._gaussians = self._get_gaussians(self._points)
+
+        gsobj = galsim._galsim.SBAdd(self._gaussians, gsparams)
+        galsim.GSObject.__init__(self, gsobj)
+
+    def calculateHLR(self):
+        """
+        calculate the half light radius of the generated points
+        """
+        pts = self._points
+        my,mx=pts.mean(axis=0)
+
+        r=np.sqrt( (pts[:,0]-my)**2 + (pts[:,1]-mx)**2)
+
+        hlr=np.median(r)
+
+        return hlr
+
+
+    @property
+    def input_half_light_radius(self):
+        """
+        getter for the input half light radius
+        """
+        return self._half_light_radius
+
+    @property
```

Comment:
Do we need this?  Doesn't the `GSObject.flux` property already cover this?  (Maybe this version is slightly faster?)